### DISCUSSION TITLE
How to avoid update override on server manual installation

### DISCUSSION THREAD

**Discussion by NguyenThiHuongVN** [TriliumNext/Trilium#4322] [Q&A]
Hi there,
I tried to install Trilium on my linux server manually via this doc.
https://github.com/zadam/trilium/wiki/Manual-server-installation

The reason to install manual is that I want to test and make change some style, function.
So, I have a question that; how to avoid next update Trilium version will not override my modified files.
EX:
- I modified function in index.js.. after update new Trilium version, I want to keep my modified.
- My solution now is take a note which files I'd modified. When update new version, I'll re-add they again. Is there better this solution? Please help me, thank you so much!

**Comment by sottey** [comment]
I am not very well versed in Github processes, but my guess would be:

1) Make a fork of Trilium on github
2) Make your changes and commit them to your trilium repo
3) When a new Trilium version comes out, I am sure there is a way to incorporate the Trilium changes to your fork (adjusting conflicts, etc)
4) update your installation with your forked repository.

It may seem like even more work than copying the changed files over the top of a new Trilium version, but it is much safer since, with your process, you will not know about any breaking changes until you have deployed and things go pear shaped.

**Reply by zerebos** [reply]
This is the way I would recommend doing it (and how I currently do it myself), and as a bonus you can still use docker this way by pointing your compose file to your local version of Trilium.

**Comment by NguyenThiHuongVN** [comment]
@sottey @rauenzi 
Thank you all for your suggestions! It's very helpful to me!
